### PR TITLE
feat([DST-1006]): Introduce a collapsible section for long checkbox groups and radios

### PR DESCRIPTION
**Pull request by sebald**
- not included: releasing the `<Collapsible>`

### CODE REVIEW

File: packages/components/src/Checkbox/CheckboxGroup.test.tsx [hunk 1]
```diff
@@ -144,3 +144,20 @@ test('horiziontal orientation style', () => {
 
   expect(presentation[0].className).toContain('flex-row gap-[1.5ch]');
 });
+
+test('don\'t show "show more" when list is too short', () => {
+  render(<CollapseAt collapseAt={100} />);
```
Comment (JimRSVX) [inline packages/components/src/Checkbox/CheckboxGroup.test.tsx:149]: maybe also test with negative numbers

File: packages/components/src/Collapsible/Context.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  value,
+  children,
+}: PropsWithChildren<{ value: CollapsibleContextProps }>) => (
+  <CollapsibleContext.Provider value={value}>
+    {children}
+  </CollapsibleContext.Provider>
+);
+
+export const useCollapsibleContext = () => {
+  const ctx = useContext(CollapsibleContext);
```
Comment (JimRSVX) [inline packages/components/src/Collapsible/Context.tsx:23]: Bad name it made me think

File: packages/components/src/Radio/RadioGroup.test.tsx [hunk 3]
```diff
@@ -1,8 +1,12 @@
...
 import { fireEvent, screen } from '@testing-library/react';
 import { vi } from 'vitest';
 import { Theme, ThemeProvider, cva } from '@marigold/system';
 import { setup } from '../test.utils';
 import { Radio } from './Radio';
+import * as stories from './Radio.stories';
+
+const { CollapseAt } = composeStories(stories);
 
 const theme: Theme = {
```
Comment (JimRSVX) [inline packages/components/src/Radio/RadioGroup.test.tsx]: refactor according to https://reservix.atlassian.net/wiki/spaces/DST/pages/3626336415/Refactoring+writing+tests+in+storybook ?

File: packages/components/src/Radio/RadioGroup.tsx [hunk 4]
```diff
@@ -132,7 +149,10 @@ const _RadioGroup = ({
         )}
       >
         <RadioGroupContext.Provider value={{ width, variant, size }}>
-          {children}
+          {visibleChildren}
+          {collapsedChildren.length > 0 ? (
+            <More>{collapsedChildren}</More>
+          ) : null}
```
Comment (JimRSVX) [inline packages/components/src/Radio/RadioGroup.tsx]: maybe make an independent component, its the same as in CheckboxGroup

File: packages/components/src/utils/children.utils.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import type { ReactNode } from 'react';
+import { Children } from 'react';
+
+export const splitChildren = (children: ReactNode, at?: number) => {
+  const c = Children.toArray(children);
```
Comment (JimRSVX) [inline packages/components/src/utils/children.utils.ts]: useful name please

File: packages/components/src/utils/children.utils.test.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+    return (
+      <>
+        <div data-testid="left">{left}</div>
+        <div data-testid="right">{right}</div>
+      </>
+    );
+  };
+
+  test('split children', () => {
```
Comment (JimRSVX) [inline packages/components/src/utils/children.utils.test.tsx:17]: you could do a parameterized test

**Comment by sebald** [comment]
Also updated the docs now. The Radio docs were kind of broken anyway

File: packages/components/src/Collapsible/Context.tsx (hunk 2 above)
Comment (sebald) [reply packages/components/src/Collapsible/Context.tsx:23]: Which variable?

File: packages/components/src/Radio/RadioGroup.tsx (hunk 4 above)
Comment (sebald) [reply packages/components/src/Radio/RadioGroup.tsx]: It's only an if/else. Creating a component for that is overkill. The more component is basically the abstraction. 

Had to refractor it now. And now I couldn't share a component between the radio and checkbox group.

File: packages/components/src/Collapsible/Context.tsx (hunk 2 above)
Comment (sebald) [reply packages/components/src/Collapsible/Context.tsx:23]: Ah you mean ctx? That is a common abbreviation 🙃

File: docs/content/components/form/radio/radio.mdx [hunk 7]
```diff
@@ -14,17 +14,21 @@ The radio button component is comprised of a set of clickable circles (the input
 <Image
```
Comment (JimRSVX) [inline docs/content/components/form/radio/radio.mdx:15]: badge is missing

Comment (sebald) [reply docs/content/components/form/radio/radio.mdx:15]: added

**Review by aromko — CHANGES_REQUESTED** [review]
Changeset is missing

**Comment by sebald** [comment]
> Changeset is missing

@aromko added
